src/App.jsx

```diff
@@ -1,14 +1,27 @@
 import './App.css';
+import ChatLog from './components/ChatLog';
+import chatMessages from './data/messages.json';
+import { useState } from 'react';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+
+  const toggleLike = (id) => {
+    const updatedMessages = messages.map((msg) =>
+      msg.id === id ? { ...msg, liked: !msg.liked } : msg
+    );
+    setMessages(updatedMessages);
+  };
+
+  const countLikes = messages.filter((msg) => msg.liked).length;
```

Comment:
Great idea to filter down the data to the liked messages and use the list length as the count!

This approach works, but you'll typically see JS devs use `reduce` to calculate a value like this:

```js
  const calculateTotalLikeCount = (chatData) => {
    return chatData.reduce((acc, chat) => {
      return chat.liked ? acc + 1 : acc;
    }, 0);
  };
```